### PR TITLE
fix: bump golang.org/x/tools to v0.1.10 to fix some go 1.18 issues

### PR DESCRIPTION
**Pull request by MathiasGr**
Description
-------------

Bump the [golang.org/x/tools](https://pkg.go.dev/golang.org/x/tools@v0.1.10) dependency to v0.1.10 to fix some issues during code gen.

Why bumping x/tools specifically? I got a hint from the ent/ent repo (see [issue](https://github.com/ent/ent/issues/2155), [PR](https://github.com/ent/contrib/pull/276/files)) that had the exact same issue recently and fixed it by doing so.

Example running mockery with go 1.18 locally before the fix:

```sh
mockery <...>
config dir: <...>
config output:<...>
2022/04/11 20:11:11 internal error: package "context" without types was imported from "<...>"
```

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

Version of Golang used when building/testing:
---------------------------------------------

- [ ] 1.11
- [ ] 1.12
- [ ] 1.13
- [ ] 1.14
- [ ] 1.15
- [ ] 1.16
- [x] 1.17
- [x] 1.18

How Has This Been Tested?
---------------------------

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Checklist
-----------

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -422,8 +423,8 @@ golang.org/x/mod v0.4.0/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
 golang.org/x/mod v0.4.1/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
 golang.org/x/mod v0.4.2/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
 golang.org/x/mod v0.5.0/go.mod h1:5OXOZSfqPIIbmVBIIKWRFfZjPR0E5r58TLhUjH0a2Ro=
-golang.org/x/mod v0.5.1 h1:OJxoQ/rynoF0dcCdI7cLPktw/hR2cueqYfjm43oqK38=
-golang.org/x/mod v0.5.1/go.mod h1:5OXOZSfqPIIbmVBIIKWRFfZjPR0E5r58TLhUjH0a2Ro=
+golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3 h1:kQgndtyPBW/JIYERgdxfwMYh3AVStj88WQTlNDi2a+o=
+golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3/go.mod h1:3p9vT2HGsQu2K1YbXdKPJLVgG5VJdoTa1poYQBtP1AY=
```
Comment (LandonTClipp) [inline go.sum:427]: We shouldn't use a dev version if possible, if we remove this is the original problem still resolved?

Comment (MathiasGr) [reply go.sum:427]: Hi there! I think it's a transitive dependency so I can't really do much about it. See https://cs.opensource.google/go/x/tools/+/refs/tags/v0.1.10:go.mod

Comment (LandonTClipp) [reply go.sum:427]: I see thanks, I will get this merged in a few hours.

**Review by LandonTClipp — APPROVED** [review]

File: go.sum (hunk 1 above)
Comment (MathiasGr) [reply go.sum:427]: Awesome thanks 🙏🏻 🤩

**Comment by hassansin** [comment]
Also got affected by this. Looking forward to the release. Thanks

**Comment by Lawrence-Dawson** [comment]
I'm also being impacted by this. Eagerly awaiting the release. Cheers

**Pull request merged**